src/field.cpp

```diff
@@ -881,6 +881,35 @@ bool map::process_fields_in_submap( submap *const current_submap,
                                 bool destroyed = fuel->burn( frd, can_spread);
 
                                 if( destroyed ) {
+									//Check what the item is made of and spawn appropriate byproducts
+									std::vector<material_id> all_mats = fuel->made_of();
+									if( all_mats.size() > 0 )
+									{
+									    units::mass fuel_weight = fuel->type->weight;
```

Comment:
Why is this using the type weight instead of the item weight `item::weigth`. Those are not necessarily the same same (e.g. for items with dynamic weight (corpses)). The function has a parameter to exclude the weight of its contents.